src/transport/builder/umessagebuilder.rs

```diff
@@ -0,0 +1,733 @@
+    ///
+    /// # Returns
+    ///
+    /// The builder.
+    ///
+    /// # Examples
+    ///
+    /// ```rust
+    /// # use up_rust::uprotocol::SerializationError;
+    /// use up_rust::uuid::builder::UUIDBuilder;
+    /// use up_rust::uprotocol::{UMessageType, UPriority, UUri};
+    /// use up_rust::transport::builder::UMessageBuilder;
+    ///
+    /// # fn main() -> Result<(), Box<dyn std::error::Error>> {
+    /// let invoked_method = UUri::try_from("my-vehicle/cabin/1/rpc.unlock_doors")?;
+    /// let reply_to_address = UUri::try_from("my-cloud/dashboard/1/rpc.response")?;
+    /// let request_id = UUIDBuilder::new().build();
+    /// let message = UMessageBuilder::response(&reply_to_address, &request_id, &invoked_method)
+    ///                     .with_comm_status(3)
+    ///                     .build()?;
+    /// assert_eq!(message.attributes.commstatus, Some(3));
+    /// # Ok(())
+    /// # }
+    /// ```
+    pub fn with_comm_status(&'a mut self, comm_status: i32) -> &'a mut UMessageBuilder {
+        self.comm_status = Some(comm_status);
+        self
+    }
+
+    /// Sets the message's request ID.
+    ///
+    /// If this builder is for a request message and the request ID is not set explicitly using this function,
+    /// a request ID will be generated when the message is being built.
+    ///
+    /// # Arguments
+    ///
+    /// * `request_id` - The ID.
+    ///
+    /// # Returns
+    ///
+    /// The builder.
+    ///
+    /// # Examples
+    ///
+    /// ```rust
+    /// # use up_rust::uprotocol::SerializationError;
+    /// use up_rust::uprotocol::{UMessageType, UPriority, UUri};
+    /// use up_rust::uuid::builder::UUIDBuilder;
+    /// use up_rust::transport::builder::UMessageBuilder;
+    ///
+    /// # fn main() -> Result<(), Box<dyn std::error::Error>> {
+    /// let method_to_invoke = UUri::try_from("my-vehicle/cabin/1/rpc.unlock_doors")?;
+    /// let reply_to_address = UUri::try_from("my-cloud/dashboard/1/rpc.response")?;
+    /// let request_id = UUIDBuilder::new().build();
+    /// let message = UMessageBuilder::request(&method_to_invoke, &reply_to_address, 5000)
+    ///                     .with_request_id(&request_id)
+    ///                     .build()?;
+    /// assert_eq!(message.attributes.reqid, Some(request_id).into());
+    /// # Ok(())
+    /// # }
+    /// ```
+    pub fn with_request_id(&mut self, request_id: &'a UUID) -> &'a mut UMessageBuilder {
+        self.request_id = Some(request_id);
+        self
+    }
+
+    /// Creates the message based on the builder's state.
+    ///
+    /// # Returns
+    ///
+    /// A message ready to be sent using [`crate::transport::datamodel::UTransport::send`].
+    ///
+    /// # Errors
+    ///
+    /// Returns an error if the properties set on the builder do not represent a valid set of [`UAttributes`].
+    pub fn build(&self) -> Result<UMessage, ValidationError> {
+        let uuid_builder = UUIDBuilder::new();
```

Comment:
IMHO this is a bit against the spirit of what is described as the [UUID in the spec](https://github.com/eclipse-uprotocol/up-spec/blob/ceb40c3e218ba29241f8cc8f10aa141a7a2d3c84/basics/uuid.adoc#2-specification), in particular w/r/t `rand_b`:
> * MUST be a 62 bits pseudo-random number generated per uE
> * MUST NOT change during the lifecycle of the uE
In other words, there should not ever be a time that a uE should be coming up with a new `rand_b` to describe itself on the fly and the `build_with()` function below should be the only available way to obtain a final `UMessage`.

WDYT?

Reply:
I had the same thought when I added the `build_with` function. If I understand you correctly, you propose to replace the original `build` function with `build_with` (and rename that to `build` then), right?